replication-api/src/main/java/com/connexta/replication/api/data/FilterIndex.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.api.data;
+
+import com.connexta.ion.replication.api.data.Metadata;
+import java.time.Instant;
+
+/** Keeps track of what to query from a {@link com.connexta.replication.api.data.Site}. */
+public interface FilterIndex extends Persistable {
+
+  /**
+   * A time which represents the modified date of the last {@link Metadata} that was put to the
+   * Queue.
+   *
+   * @return the {@link Instant}, or {@code null} if no metadata has attempted to be replicated
+   */
+  Instant getModifiedSince();
```

Comment:
❗️ ❓ Why would such an object exist if we haven't tried to replicate yet? I would make it a required field and not support it being `null`

Reply:
Would there be a case with new fields in the future where we would instantiate an index before the first query?

Reply:
🤔 Maybe. So let's not make it require but do make it Optional<Instance> instead of returning `null`. The pojo would still return `null`